staker-rewards/lib/rewards-calculator.js

```diff
@@ -26,12 +26,24 @@ export default class RewardsCalculator {
     const operatorsRewardsFactors = []
 
     for (const operatorParameters of operatorsParameters) {
-      const { keepStaked, ethTotal } = operatorParameters.operatorAssets
+      let ethTotal = operatorParameters.operatorAssets.ethTotal
+      const { keepStaked } = operatorParameters.operatorAssets
 
-      const requirementsViolations = this.checkRequirementsViolations(
+      let requirementsViolations = this.checkRequirementsViolations(
         operatorParameters
       )
 
+      if (
+        requirementsViolations.includes("poolRequirementFulfilledAtStart") &&
+        (await this.isUndelegatingOperator(operatorParameters.operator))
+      ) {
+        requirementsViolations = requirementsViolations.filter(
+          (item) => item !== "poolRequirementFulfilledAtStart"
+        )
+
+        ethTotal = operatorParameters.operatorAssets.ethBonded
+      }
```

Comment:
This condition does not cover the situation when someone is undelegating but is still in the sortition pool because the sortition pool had no chance to remove that operator yet.

The way it should work:
- if you are undelegating, you are not required to be in the sortition pool
- if you are undelegating, only ETH_bonded counts to ETH_total

Right now, it is:
- if you are undelegating, you are not required to be in the sortition pool
- if you are undelegating and if you are not in the sortition pool, only ETH_bonded counts to ETH_total

Reply:
Good catch. I will change.